CMakeLists.txt

```diff
@@ -419,49 +457,51 @@ tr_add_external_auto_library(EVENT2 libevent event
         -DEVENT__DISABLE_SAMPLES:BOOL=ON
         -DEVENT__LIBRARY_TYPE:STRING=STATIC)
 
-tr_add_external_auto_library(NATPMP libnatpmp natpmp)
+tr_add_external_auto_library(NATPMP libnatpmp natpmp
+    TARGET natpmp::natpmp)
 if(NOT USE_SYSTEM_NATPMP)
-    set(NATPMP_DEFINITIONS NATPMP_STATICLIB)
+    target_compile_definitions(natpmp::natpmp
+        INTERFACE
+            NATPMP_STATICLIB)
 endif()
 
 tr_add_external_auto_library(MINIUPNPC miniupnpc miniupnpc
+    TARGET miniupnpc::libminiupnpc
     CMAKE_ARGS
         -DUPNPC_BUILD_STATIC=ON
         -DUPNPC_BUILD_SHARED=OFF
         -DUPNPC_BUILD_TESTS=OFF)
```

Comment:
In my opinion, it's a bad practice to have closing parentheses on the same line as the last element:
- it's harder to see when the list ends
- it's causing conflicts in versioning, each time we add or remove an item the end of a list

The recommended modern way to have the closing parentheses alone on their line:
```
        -DUPNPC_BUILD_STATIC=ON
         -DUPNPC_BUILD_SHARED=OFF
         -DUPNPC_BUILD_TESTS=OFF
    )
```

Reply:
I'm not aware of any "recommended modern way". If you have a prooflink, please provide it. I remember reading through a couple of bugreports touching the topic but that's about it.

A style is just a style. Some people like it, some don't. CMake's own documentation has it both ways, sometimes within the same article or even within the same code sample. Same goes for their standard modules.

Not having a consistent style is worse than having a style some people like less then others. The only thing I regret is that there's no good automated formatting tool to enforce it.

Reply:
Regardless of what style the project goes with, maybe cmake-lint could be plugged in as a precommit hook to ensure that it's used?

Reply:
I'll invest some more time in automation research but don't get your hopes up just yet.